### ISSUE TITLE
Use in multi-process environment?

### ISSUE THREAD

**Issue by dinkelk** [coleifer/unqlite-python#46]
I have multiple processes reading/writing from a database and keep getting the following exception:

```
unqlite.UnQLiteError: b'Another process or thread hold the requested lock\nAnother process or thread have a reserved lock on this database\n'
```

[Here](https://unqlite.org/c_api_const.html) it says that `unqlite` is thread-safe with the `UNQLITE_ENABLE_THREADS` flag turned on during compilation. Does this python package use a compiled version of `unqlite` with that flag enabled? If not, how would I create a version that does?

**Comment by coleifer** [comment]
Do yourself a favor and use sqlite.

I don't think you need that compiler flag as long as each thread has it's own connection.

You can't, however, use multiple processes to the best of my knowledge.

**Comment by dinkelk** [comment]
For anyone else that finds this issue. I tried implementing sqlite and [sqlitedict](https://github.com/RaRe-Technologies/sqlitedict) in my application, but they were too slow for my requirements. I ended up using unqlite-python, as before, but provided multi-process mutual exclusion by wrapping simultaneous writes in locks provided by the python [filelock](https://filelock.readthedocs.io/en/latest/) module.
